kpi-queries/src/main/java/com.c12e.cortex.examples.aggregate/KPIQueries.java

```diff
@@ -0,0 +1,288 @@
+
+            return new ProfileSchema(
+                    node.get("project").asText(),
+                    node.get("name").asText(),
+                    node.has("title") ? node.get("title").asText(null) : null,
+                    node.has("description") ? node.get("description").asText(null) : null,
+                    ctxt.readTreeAsValue(node.get("names"), ProfileNames.class),
+                    ctxt.readTreeAsValue(node.get("primarySource"), DataSourceSelection.class),
+                    Arrays.asList(ctxt.readTreeAsValue(node.get("joins"), JoinSourceSelection[].class)),
+                    node.has("userId") ? node.get("userId").asText(null) : null,
+                    attributes,
+                    Arrays.asList(ctxt.readTreeAsValue(node.get("attributeTags"), AttributeTag[].class))
+            );
+        }
+    }
+
+    protected <T> T getOrDefault(Supplier<T> function, T defaultValue) {
+        try {
+            T value = function.get();
+            if (value == null) {
+                return defaultValue;
+            }
+            return value;
+
+        } catch (NullPointerException | NotFoundException e) {
+            return defaultValue;
+        }
+    }
+
+    @Override
+    public void run() {
+        var sessionExample = new SessionExample();
+        CortexSession cortexSession = sessionExample.getCortexSession();
+        String profilesBucket = cortexSession.getContext().getSparkSession().conf().getAll().get("spark.cortex.storage.bucket.profiles").get();
+
+
+        String dataSourceConfig = "{\n" +
+                "                \"project\": \"" + project + "\",\n" +
+                "                \"attributes\": [\n" +
+                "                  \"timeOfExecution\",\n" +
+                "                  \"value\",\n" +
+                "                  \"startDate\",\n" +
+                "                  \"endDate\",\n" +
+                "                  \"windowDuration\"\n" +
+                "                ],\n" +
+                "                \"connection\": {\n" +
+                "                  \"name\": \"KPI-" + name + "\"\n" +
+                "                },\n" +
+                "                \"description\": \""+description+"\",\n" +
+                "                \"kind\": \"batch\",\n" +
+                "                \"name\": \"KPI-" + name + "\",\n" +
+                "                \"primaryKey\": \"timeOfExecution\",\n" +
+                "                \"title\": \"" + name + "\"\n" +
+                "              }";
+
+        String connectionConfig = "{\n" +
+                "                \"project\": \"" + project + "\",\n" +
+                "                \"name\": \"KPI-" + name + "\",\n" +
+                "                \"title\": \"" + name + "\",\n" +
+                "                \"connectionType\": \"gcs\",\n" +
```

Comment:
Does this limit the example to being used with a `gcs` connection? (I don't remember seeing anything about that in the README).

I'm a little concerned this example might be setup with a focus for a specific use case.